src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorageService.cs

```diff
@@ -64,9 +64,29 @@ protected override string GetDatabaseFilePath(string workingFolderPath)
                 sqlStorage.Initialize(solution);
 
             }
-            catch (Exception)
+            catch (Exception exception)
             {
-                sqlStorage?.Dispose();
+                if (IsNativeLibraryResolutionException(exception))
```

Comment:
rather than doing this way, can we add explicit check for those dependencies? basically, rather than doing this implicitly through static constructor (http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces.Desktop/Workspace/SQLite/SQLitePersistentStorage.cs,96)

make an explicit static API for it such as bool SQLitePersistentStorage.InitLibraries(). which will run the LoadLibrary once and cache the result. and next time it is called, if it just return true or false.

and in TryOpenDatabase, before TryGetDatabaseOwnership, it calls SQLitePersistentStorage.InitiLibrary and if that return false, return storage = null and return false like what dbOwnershipLock does.

I think that is more explicit check and follow pattern already in this method.

...

if you are worry about same thing repeatedly get called, then, I think that check should move up to here.
http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces.Desktop/Workspace/Storage/PersistenceStorageServiceFactory.cs,24

since the condition where that LoadLibrary will fail is process wide, not per workspace or solution. the factory itself should check such condition and return 
http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Host/PersistentStorage/NoOpPersistentStorageService.cs,0ec32e8cdd068f49

since in the process, SQLite will be never supported.

Reply:
Yeah, that's a better solution. This PR ended up being bigger than I initially imagined 😛 

I'll open a PR writing it way.